Add metamonitoring similiar to GET (#5376)
grafana/tempo: docs/sources/tempo/shared/metamonitoring.md

---
headless: true
description: Shared file for metamonitoring GET and Tempo.
labels:
  products:
    - enterprise
    - oss
---

[//]: # 'This file documents the best practices for monitoring for Tempo.'
[//]: # 'This shared file is included in these locations:'
[//]: # '/tempo/docs/sources/tempo/operations/monitor/set-up-monitoring.md'
[//]: #
[//]: # 'If you make changes to this file, verify that the meaning and content are not changed in any place where the file is included.'
[//]: # 'Any links should be fully qualified and not relative: /docs/grafana/ instead of ../grafana/.'

> Metamonitoring for Tempo is handled by the [Grafana Kubernetes Helm chart](https://github.com/grafana/k8s-monitoring-helm) (>=v2.1). Using Grafana Alloy to collect metrics and logs.

Reviewer: ```suggestion
Metamonitoring for Tempo is handled by the [Grafana Kubernetes Helm chart](https://github.com/grafana/k8s-monitoring-helm) (>=v2.1).  The Helm chart uses Grafana Alloy to collect metrics and logs.
```

> 

Reviewer: At the end of this procedure, we instruct people to log into Grafana Cloud to check their stats. What if they are using Grafana open source and not Cloud? 

This is one difference for GET vs Tempo. GET requires Cloud while Tempo users can specify Grafana OSS or Cloud. 

Also, when we're in the Tempo docs, we try to minimize any emphasis on Cloud.

Author: I have removed any specific references to cloud I added. I did rework the URL help text in the values.yaml so double check what you like there. In Alloy we found people left off things all the time. IE only put something like `https://loki.grafana.net`

Reviewer: I've often wondered how many times people forget to update the key fields in the yaml files and then wonder why it doesn't work.

## Steps

1. Add the Grafana Helm Chart repository, or update if already added. This will be using the [Grafana Kubernetes Helm chart](https://github.com/grafana/k8s-monitoring-helm) and the `values.yml` file will be to set parameters in the Helm chart.

> ```

Author: Most people will, combining them.

helm repo add grafana https://grafana.github.io/helm-charts
```

```
helm repo update
```

> 1. Generate `values.yml`.

Author: In the current directory by default or as long as you reference the full path in helm install command.

    ```yaml
    cluster:
    name: traces # Name of the cluster, this will populate the cluster label

    integrations:
    tempo:
        instances:
        - name: "traces" # This is the name for the instance label that will be reported.
            namespaces:
            - traces # This is the namespace that will be searched for tempo instances, change this accordingly
            metrics:
            enabled: true
            portName: prom-metrics
            logs:
            enabled: true
            labelSelectors:
            app.kubernetes.io/name: tempo

    alloy:
        name: "traces-monitoring"

    destinations:
    - name: "metrics"
        type: prometheus
        url: "<url>" # URL for prometheus, this can be found under your Grafana Cloud account and should look similiar to "https://prometheus-us-central1.grafana.net/api/prom/push", if using local Prometheus use that url
        auth:
        type: basic
        username: "<username>"
        password: "<password>"

    - name: "logs"
        type: loki

>         url: "<url>" # URL for loki, this can be found under your Grafana Cloud account and should look similiar to "https://logs-prod-us-central1.grafana.net/loki/api/v1/push", if using local Loki use that url

Reviewer: ```suggestion
        url: "<url>" # URL for Loki. Located under your Grafana Cloud account and should look similar to "https://logs-prod-us-central1.grafana.net/loki/api/v1/push". If you're using local Loki, use that URL.
```

        auth:
        type: basic
        username: "<username>" 
        password: "<password>"

    alloy-metrics:
    enabled: true

    podLogs:
    enabled: true
    gatherMethod: kubernetesApi
    namespaces: [traces] # Set to namespace from above under instances.
    collector: alloy-singleton

    alloy-singleton:
    enabled: true

    alloy-metrics:

>     enabled: true # This will send Grafana Alloy metrics to ensure the monitoring is working properly.

Reviewer: ```suggestion
    enabled: true # Sends Grafana Alloy metrics to ensure the monitoring is working properly.
```

    ```

> 1. Install the Helm chart using the following command create Grafana Alloy instances to scrape metrics and logs:

Reviewer: ```suggestion
1. Install the Helm chart using the following command to create Grafana Alloy instances to scrape metrics and logs:
```

    ```
    helm install k8s-monitoring grafana/k8s-monitoring \
    --namespace monitoring \
    --create-namespace \
    -f values.yml
    ```

> 
> 
> 1. View Logs and Metrics
> 
>     Log into Grafana Cloud. Select metrics drilldown and select `cluster=<cluster.name>` where cluster.name is the name specified in the values.yml. Do the same for logs drilldown.

Reviewer: Do you have to use Grafana Cloud or can you use local Grafana? 
```suggestion

1. Verify that data is being sent to Grafana or Grafana Cloud. 
    - Log into Grafana or Grafana Cloud. 
    - Select Metrics Drilldown and select `cluster=<cluster.name>` where `cluster.name` is the name specified in the `values.yml` file. 
    - Do the same for Logs Drilldown.
```

---
headless: true
description: Shared file for metamonitoring GET and Tempo.
labels:
  products:
    - enterprise
    - oss
---

[//]: # 'This file documents the best practices for monitoring for Tempo.'
[//]: # 'This shared file is included in these locations:'
[//]: # '/tempo/docs/sources/tempo/operations/monitor/set-up-monitoring.md'
[//]: #
[//]: # 'If you make changes to this file, verify that the meaning and content are not changed in any place where the file is included.'
[//]: # 'Any links should be fully qualified and not relative: /docs/grafana/ instead of ../grafana/.'

Metamonitoring for Tempo is handled by the [Grafana Kubernetes Helm chart](https://github.com/grafana/k8s-monitoring-helm) (>=v2.1). Metamonitoring can be used with both microservices and single binary deployments of Tempo.

The Helm chart configures Grafana Alloy to collect metrics and logs.

## Steps

This procudure uses the [Grafana Kubernetes Helm chart](https://github.com/grafana/k8s-monitoring-helm) and the `values.yml` file sets parameters in the Helm chart.

1. Add the Grafana Helm Chart repository, or update, if already added. 

> 

Reviewer: Should this example be indented underneath? OTherwise, it will probably reset the numbered list.

    ```yaml
    cluster:
    name: traces # Name of the cluster. This populates the cluster label.

    integrations:
    tempo:
        instances:
        - name: "traces" # This is the name for the instance label that reports.
            namespaces:
            - traces # This is the namespace that is searched for Tempo instances. Change this accordingly.
            metrics:
            enabled: true
            portName: prom-metrics
            logs:
            enabled: true
            labelSelectors:
            app.kubernetes.io/name: tempo

    alloy:
        name: "traces-monitoring"

    destinations:
    - name: "metrics"
        type: prometheus
        url: "<url>" # URL for Prometheus. Should look similar to "https://<prometheus host>/api/prom/push".
        auth:
        type: basic
        username: "<username>"
        password: "<password>"

    - name: "logs"
        type: loki
        url: "<url>" # URL for Loki. Should look similar to "https://<loki host>/loki/api/v1/push".
        auth:
        type: basic
        username: "<username>" 
        password: "<password>"

    alloy-metrics:

>     enabled: true

Author: My formatting got screwed up somewhere need to fix that.
